File: src/theme/colors.ts
/**
 * Centralized color palette for the MaaS app.
 * All colors used across the app should be referenced from here.
 */

export const Colors = {
  // ─── Brand (legacy dark theme) ────────────────────────
  primary: '#4776E6',
  primaryLight: '#8E54E9',
  accent: '#FF416C',
  accentWarm: '#FF4B2B',
  success: '#00E676',
  info: '#40C4FF',
  warning: '#FFD54F',

  // ─── Salmon Palette ───────────────────────────────────
  salmon: '#FF7B5F',
  salmonLight: '#FF9B85',
  salmonSoft: '#FFB5A3',
  coralIntense: '#FF6B4A',
  peach: '#FFC9BD',
  charcoal: '#2D3339',
  grayNeutral: '#6B7280',
  successGreen: '#10B981',

  // ─── Backgrounds ─────────────────────────────────────
  bgDark: '#0F2027',
  bgDarkMid: '#203A43',
  bgDarkDeep: '#101522',
  bgCard: '#1A1A24',
  bgBlack: '#000000',
  bgWhite: '#FFFFFF',
  bgLightGray: '#F7F7F9',
  bgSurface: 'rgba(255,255,255,0.05)',
  bgSurfaceLight: 'rgba(255,255,255,0.1)',
  bgSurfaceHover: 'rgba(255,255,255,0.15)',
  bgSurfaceMedium: 'rgba(255,255,255,0.2)',
  bgSurfaceBright: 'rgba(255,255,255,0.3)',
  bgPrimarySubtle: 'rgba(71, 118, 230, 0.1)',
  bgDangerSubtle: 'rgba(255, 65, 108, 0.1)',

  // ─── Borders ──────────────────────────────────────────
  borderSubtle: 'rgba(255,255,255,0.05)',
  borderLight: 'rgba(255,255,255,0.1)',
  borderMedium: 'rgba(255,255,255,0.15)',
  borderBright: 'rgba(255,255,255,0.3)',
  borderDanger: 'rgba(255, 65, 108, 0.3)',
  borderLightGray: '#E5E7EB',

  // ─── Text ─────────────────────────────────────────────
  textPrimary: '#fff',
  textSecondary: '#8892B0',
  textMuted: '#B0BEC5',
  textOnSurface: '#E0E6ED',
  textDanger: '#FF416C',
  textDimmed: 'rgba(255,255,255,0.6)',
  textLabel: 'rgba(255,255,255,0.7)',

  // ─── Light / Dark theme (for ThemedText/ThemedView) ──
  light: {
    text: '#000000',
    background: '#ffffff',
    backgroundElement: '#F0F0F3',
    backgroundSelected: '#E0E1E6',
    textSecondary: '#60646C',
  },
  dark: {
    text: '#ffffff',
    background: '#000000',
    backgroundElement: '#212225',
    backgroundSelected: '#2E3135',
    textSecondary: '#B0B4BA',
  },
} as const;

export type ThemeColor = keyof typeof Colors.light & keyof typeof Colors.dark;

// ─── Gradient presets ─────────────────────────────────────
export const Gradients = {
  main: ['#0F2027', '#203A43', '#2C5364'] as const,
  dark: ['#101522', '#1A1A24', '#0F2027'] as const,
  darkFlat: ['#000000', '#1A1A24'] as const,
  primary: ['#4776E6', '#8E54E9'] as const,
  accent: ['#FF416C', '#FF4B2B'] as const,
  info: ['#00B4DB', '#0083B0'] as const,
  surface: ['rgba(255,255,255,0.1)', 'rgba(255,255,255,0.05)'] as const,
  salmonCard: ['#FF7B5F', '#FF9B85', '#FFB5A3'] as const,
  salmonButton: ['#FF7B5F', '#FF6B4A'] as const,
} as const;

File: lib/core/theme/colors.dart
import 'package:flutter/material.dart';

/// Centralized color palette for the Guayaba/MaaS app.
class AppColors {
  AppColors._();

  // ─── Brand (legacy dark theme) ─────────────────
  static const primary = Color(0xFF4776E6);
  static const primaryLight = Color(0xFF8E54E9);
  static const accent = Color(0xFFFF416C);
  static const accentWarm = Color(0xFFFF4B2B);
  static const success = Color(0xFF00E676);
  static const info = Color(0xFF40C4FF);
  static const warning = Color(0xFFFFD54F);

  // ─── Salmon Palette ────────────────────────────
  static const salmon = Color(0xFFFF7B5F);
  static const salmonLight = Color(0xFFFF9B85);
  static const salmonSoft = Color(0xFFFFB5A3);
  static const coralIntense = Color(0xFFFF6B4A);
  static const peach = Color(0xFFFFC9BD);
  static const charcoal = Color(0xFF2D3339);
  static const grayNeutral = Color(0xFF6B7280);
  static const successGreen = Color(0xFF10B981);

  // ─── Backgrounds ───────────────────────────────
  static const bgDark = Color(0xFF0F2027);
  static const bgDarkMid = Color(0xFF203A43);
  static const bgDarkDeep = Color(0xFF101522);
  static const bgCard = Color(0xFF1A1A24);
  static const bgBlack = Color(0xFF000000);
  static const bgWhite = Color(0xFFFFFFFF);
  static const bgLightGray = Color(0xFFF7F7F9);
  static const bgSurface = Color(0x0DFFFFFF);
  static const bgSurfaceLight = Color(0x1AFFFFFF);

  // ─── Borders ───────────────────────────────────
  static const borderSubtle = Color(0x0DFFFFFF);
  static const borderLight = Color(0x1AFFFFFF);
  static const borderLightGray = Color(0xFFE5E7EB);

  // ─── Text ──────────────────────────────────────
  static const textPrimary = Color(0xFFFFFFFF);
  static const textSecondary = Color(0xFF8892B0);
  static const textMuted = Color(0xFFB0BEC5);
  static const textDanger = Color(0xFFFF416C);
}

/// Gradient presets
class AppGradients {
  AppGradients._();

  static const main = [Color(0xFF0F2027), Color(0xFF203A43), Color(0xFF2C5364)];
  static const dark = [Color(0xFF101522), Color(0xFF1A1A24), Color(0xFF0F2027)];
  static const darkFlat = [Color(0xFF000000), Color(0xFF1A1A24)];
  static const primary = [Color(0xFF4776E6), Color(0xFF8E54E9)];
  static const accentGrad = [Color(0xFFFF416C), Color(0xFFFF4B2B)];
  static const salmonCard = [Color(0xFFFF7B5F), Color(0xFFFF9B85), Color(0xFFFFB5A3)];
  static const salmonButton = [Color(0xFFFF7B5F), Color(0xFFFF6B4A)];
}
